### PR TITLE
LPS-125660 Improve REST Builder generation to consider external references from inner properties inside components

### PR DESCRIPTION
**Pull request by javierdearcos**

### CODE REVIEW

**Comment by javierdearcos** [comment]
@jkappler This should unblock you and let you use Segment type inside Experience DTO instead of Object.

Take into account that you should add the reference to Headless Admin User API from the Headless Delivery API to work

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java [hunk 1]
```diff
@@ -449,25 +461,68 @@ public static boolean hasHTTPMethod(
...
-
-				externalReferences.add(reference);
+			Queue<Map<String, Schema>> queue = new LinkedList<>();
+
+			queue.add(Collections.singletonMap("content", contentSchema));
+
+			Map<String, Schema> map = null;
+			Set<String> visited = new HashSet<>();
+
+			while ((map = queue.poll()) != null) {
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java:485]: Can we extract this somewhere? I think it's the third time with the same structure.

**Review by nhpatt — APPROVED** [review]

**Comment by nhpatt** [comment]
ci:forward

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java (hunk 1 above)
Comment (javierdearcos) [reply modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java:485]: In this case, I think that it is not so similar to the other 2 (much similar between them) and I think would be difficult to generalize for the 3 cases. May be we can give it a look to the other 2 to refactor them

**Pull request closed without merging**